internal/runbits/runtimemsg.go

```diff
@@ -3,61 +3,97 @@ package runbits
 // Progress bar design
 //
 import (
+	"context"
 	"fmt"
+	"strconv"
+	"time"
 
-	"github.com/go-openapi/strfmt"
+	"github.com/thoas/go-funk"
+	"github.com/vbauerster/mpb/v6"
 
+	"github.com/ActiveState/cli/internal/errs"
+	"github.com/ActiveState/cli/internal/locale"
+	"github.com/ActiveState/cli/internal/logging"
 	"github.com/ActiveState/cli/internal/output"
+	"github.com/ActiveState/cli/internal/runbits/progressbar"
 	"github.com/ActiveState/cli/pkg/platform/runtime/artifact"
-	"github.com/ActiveState/cli/pkg/platform/runtime/setup"
+	"github.com/ActiveState/cli/pkg/platform/runtime/setup/events"
 )
 
 type RuntimeMessageHandler struct {
 	out output.Outputer
 }
 
 func NewRuntimeMessageHandler(out output.Outputer) *RuntimeMessageHandler {
-	return &RuntimeMessageHandler{out: out}
+	return &RuntimeMessageHandler{out}
 }
 
-func (r RuntimeMessageHandler) BuildStarting(total int) {
-	r.out.Notice(fmt.Sprintf("Build Starting: %d", total))
+func (rmh *RuntimeMessageHandler) ChangeSummary(artifacts artifact.ArtifactRecipeMap, requested artifact.ArtifactChangeset, changed artifact.ArtifactChangeset) error {
+	// currently we only print a change summary if we are adding exactly ONE package
+	if len(requested.Added) != 1 {
+		return nil
+	}
+
+	ar, ok := artifacts[requested.Added[0]]
+	if !ok {
+		return errs.New("Did not find requested artifact in ArtifactRecipeMap")
+	}
+
+	// the added (direct dependencies) of this artifact that are actually new in this project
+	addedDependencies := funk.Join(ar.Dependencies, changed.Added, funk.InnerJoin).([]artifact.ArtifactID)
+
+	rmh.out.Notice("")
+	rmh.out.Notice(locale.Tl(
+		"changesummary_title",
+		"[NOTICE]{{.V0}}[/RESET] includes {{.V1}} dependencies, for a combined total of {{.V2}} new dependencies.",
+		ar.Name, strconv.Itoa(len(addedDependencies)), strconv.Itoa(len(changed.Added)),
+	))
+	for i, dep := range addedDependencies {
+		depMapping, ok := artifacts[dep]
+		if !ok {
+			logging.Error("Could not find dependency %s in artifactsMap", dep)
+			continue
+		}
+		var depCount string
+		recDeps := artifact.RecursiveDependenciesFor(dep, artifacts)
+		filteredRecDeps := funk.Join(recDeps, changed.Added, funk.InnerJoin).([]artifact.ArtifactID)
+		if len(filteredRecDeps) > 0 {
+			depCount = locale.Tl("ingredient_dependency_count", " ({{.V0}} dependencies)", strconv.Itoa(len(filteredRecDeps)))
+		}
+		prefix := "├─"
+		if i == len(addedDependencies)-1 {
+			prefix = "└─"
+		}
+		rmh.out.Notice(fmt.Sprintf("  [DISABLED]%s[/RESET] %s%s", prefix, depMapping.Name, depCount))
+	}
+	return nil
 }
 
-func (r RuntimeMessageHandler) BuildFinished() {
-	r.out.Notice(fmt.Sprintf("Build Finished"))
+// HandleUpdateEvents prints output based on runtime events received on the eventCh
+func (rmh *RuntimeMessageHandler) HandleUpdateEvents(eventCh <-chan events.BaseEventer) {
```

Comment:
This is essentially creating a whole new message handler inside the thing that is already a message handler. Please simplify.